Jenkinsfile

```diff
@@ -241,6 +245,26 @@ void runTest(Integer TEST_ID) {
     }
 }
 
+void checkE2EIgnoreFiles() {
+    def e2eignoreFile = ".e2eignore"
+    if (fileExists(e2eignoreFile)) {
+        def excludedFiles = readFile(e2eignoreFile).split('\n').collect{it.trim()}
+        def changedFiles = sh(script: "git diff --name-only origin/${env.CHANGE_TARGET}", returnStdout: true).trim().split('\n')
+        echo "Excluded files: ${excludedFiles}"
+        echo "Changed files: ${changedFiles}"
+
+        def excludedFilesRegex = excludedFiles.collect{it.replace("**", ".*").replace("*", "[^/]*")}
+        def excludedFileList = changedFiles.every{changed -> excludedFilesRegex.any {regex -> changed ==~ regex}}
+
+        if (excludedFileList) {
+            currentBuild.result = 'ABORTED'
```

Comment:
does it mean the build will fail? if yes, it'll be impossible to merge PRs without superpowers

Reply:
It will be in an Aborted state. Yes you're right. I will find another way to pass the pipeline run without aborting it.
![Screenshot_2024-12-06_12-19-47](https://github.com/user-attachments/assets/377eb879-f304-4519-9c97-0a8046a56dbc)